src/rgw/rgw_crypt.cc

```diff
@@ -1022,15 +1022,20 @@ int rgw_s3_prepare_encrypt(struct req_state* s,
         get_crypt_attribute(s->info.env, parts, X_AMZ_SERVER_SIDE_ENCRYPTION_CUSTOMER_ALGORITHM);
     if (! req_sse_ca.empty()) {
       if (req_sse_ca != "AES256") {
+        ldout(s->cct, 5) << "ERROR: Invalid value for header"
+                         << "x-amz-server-side-encryption-customer-algorithm"
```

Comment:
@theanalyst I like the debug messages you introduced. I think encryption failures in L1170 L1180 L1194 should also be noted.